apps/storeutl.c

```diff
@@ -458,18 +460,37 @@ static int process(const char *uri, const UI_METHOD *uimeth, PW_CB_DATA *uidata,
                 PEM_write_bio_PUBKEY(out, OSSL_STORE_INFO_get0_PUBKEY(info));
             break;
         case OSSL_STORE_INFO_PKEY:
-            if (text)
-                EVP_PKEY_print_private(out, OSSL_STORE_INFO_get0_PKEY(info),
-                                       0, NULL);
-            if (!noout)
-                PEM_write_bio_PrivateKey(out, OSSL_STORE_INFO_get0_PKEY(info),
-                                         NULL, NULL, 0, NULL, NULL);
+            {
+                EVP_PKEY *pkey = OSSL_STORE_INFO_get0_PKEY(info);
+
+                if (pkey == NULL) {
+                    items--;
+                    BIO_printf(bio_err, "Key not actually present\n");
+                    ret++;
```

Comment:
Uhmmmmmmmm...  when does this happen?

Reply:
I cannot recall in detail (I added this case exactly half a year ago) in a commit with the message
```
    apps/storeutl: Fix in case PKCS#12 input has no key or no user cert
```
but I think this may happen if for some reason the PKCS#12 input has a cert but not a key.
At least this defensive check should not hurt.

Reply:
I've just tried this out with a PKCS#12 file not having a key.
It looks like in `case OSSL_STORE_INFO_PKEY:` this cannot happen, so I removed that check again.